TezosChain.ts

```diff
@@ -734,62 +734,67 @@ export class TezosChain extends pulumi.ComponentResource {
         },
         { provider: this.provider }
       )
-      if (name.includes("dailynet")) {
 
-        // Define a custom awaiter function
-        const waitForDalIps = async () => {
-          while (true) {
-            // Get the status of the service from the live object
-            const dalBootstrapLbStatus = await dalBootstrapLb.status;
-            const dal1LbStatus = await dal1Lb.status;
-
-            // Check if the service is ready and the IP address is available
-            if (dalBootstrapLbStatus && dalBootstrapLbStatus.loadBalancer && dalBootstrapLbStatus.loadBalancer.ingress[0].ip &&
-              dal1LbStatus && dal1LbStatus.loadBalancer && dal1LbStatus.loadBalancer.ingress[0].ip) {
-              break;
-            }
-
-            // Wait for 10 seconds before the next check
-            await new Promise((resolve) => setTimeout(resolve, 10000));
+      params.helmValues.node_config_network.dal_config.bootstrap_peers = [
+        `${dalBootstrapP2pFqdn}:11732`,
+      ]
+
+      // Wait for DAL LB to get their IP addresses.
+      let dal1LbIp;
+      let dalBootstrapLbIp;
+      const waitForDalIps = async () => {
+        while (true) {
+          // Get the status of the service from the live object
+
+          dal1LbIp = (k8s.core.v1.Service.get(name, `${name}-dal-dal1`).status)?.loadBalancer?.ingress?.[0]?.ip;
+          dalBootstrapLbIp = (k8s.core.v1.Service.get(name, `${name}-dal-bootrstrap`).status)?.loadBalancer?.ingress?.[0]?.ip;
+
+          // If the IP address is available, the service is ready, hence break
+          if (dal1LbIp && dalBootstrapLbIp) {
+            break;
           }
+
         };
+
+        // Wait for 10 seconds before the next check
+        await new Promise((resolve) => setTimeout(resolve, 10000));
+
+      }
+      if (name.includes("dailynet")) {
         waitForDalIps();
```

Comment:
I believe this needs to be awaited too otherwise the nodejs execution continues to run to the next line before the ip's are resolved. The custom awaiter code does [this](https://gist.github.com/lukehoban/fd0355ed5b82386bd89c0ffe2a3c916a#file-waitforjob-ts-L40) which should work. In our case we'll need `pulumi.all` to await multiple things at once

Reply:
but I am not within an `async`. shall I made the entire section where this resides an async?

Reply:
I would first try what the reference code does. something like `const jobDone = job.metadata.apply(metadata => waitForJob(metadata));
`